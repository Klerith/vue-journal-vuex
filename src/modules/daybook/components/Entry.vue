<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id }})"
  >
      <!-- Titilo -->
      <div class="entry-title d-flex">
          <span class="text-success fs-5 fw-bold">{{ day }}</span>
          <span class="mx-1 fs-5">{{ month }}</span>
          <span class="mx-2 fw-light">{{ yearDay }}</span>
      </div>

      <div class="entry-description">
          {{ shortText }}
      </div>
  </div>
</template>

<script>
const months = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio','Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre']
const days   = ['Domingo','Lunes','Martes','Miércoles','Jueves','Viernes','Sábado']


export default {
    props: {
        entry: {
            type: Object,
            required: true
        }
    },
    computed: {
        shortText() {
            return ( this.entry.text.length > 130 )
                ? this.entry.text.substring(0,130) + '...'
                : this.entry.text
        },
        day() {
            const date = new Date( this.entry.date )
            return date.getDate()
        },
        month() {
            const date = new Date( this.entry.date )
            return months[ date.getMonth() ]
        },
        yearDay() {
            const date = new Date( this.entry.date )
            return `${ date.getFullYear() }, ${ days[ date.getDay() ] }`
        }
    }
}
</script>

<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }

    .entry-description {
        font-size: 12px;
    }
}

</style>