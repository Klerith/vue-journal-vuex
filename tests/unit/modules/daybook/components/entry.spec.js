import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Pruebas en Entry Component", () => {

    const mockRouter = {
        push: jest.fn()
    };

    const wrapper = shallowMount(Entry, {
        props: {
        entry: journalState.entries[0]
        },
        global: {
        mocks: {
            $router: mockRouter
        }
        }
    });

    test("debe de hacer match con el snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("debe de redireccionar al hacer click en el entry-container", () => {

        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith({
                name: 'entry', 
                params: {
                    id: journalState.entries[0].id
                }
            })

    });

    test("pruebas en las propiedades computadas", () => {

        expect( wrapper.vm.day ).toBe(23)
        expect( wrapper.vm.month ).toBe('Julio')
        expect( wrapper.vm.yearDay ).toBe('2021, Viernes')

    });
});
