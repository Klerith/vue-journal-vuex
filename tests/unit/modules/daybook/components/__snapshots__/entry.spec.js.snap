// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titilo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">Julio</span><span class="mx-2 fw-light">2021, Viernes</span></div>
  <div class="entry-description">Hola mundo desde mock data</div>
</div>
`;
